unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCSocialApiBridge.cs

```diff
@@ -0,0 +1,301 @@
+            //         GetFriendRequestId(friendRequest.User.Address, friendRequest.CreatedAt), friendRequest.CreatedAt, ownUserProfile.userId, friendRequest.User.Address, friendRequest.Message));
+            // }
+
+            return new FriendshipInitializationMessage()
+            {
+                // totalReceivedRequests = requestEvents.Incoming.Items.Count,
+                totalReceivedRequests = 0
+            };
+        }
+
+        public async UniTask ListenToFriendEvents(CancellationToken cancellationToken = default)
+        {
+            var ownUserProfile = UserProfile.GetOwnUserProfile();
+
+            await UniTask.WaitUntil(() => ownUserProfile.userId != null, PlayerLoopTiming.Update, cancellationToken);
+
+            var stream = socialClient.SubscribeFriendshipEventsUpdates(new Payload() { SynapseToken = "" });
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            await foreach (var item in stream.WithCancellation(cancellationToken))
+            {
+                foreach (var friendshipEvent in item.Events)
+                {
+                    // var action = ToFriendshipAction(friendshipEvent.BodyCase);
+                    //
+                    // OnReceivedFriendshipEvent?.Invoke(new FriendshipUpdateStatusMessage()
+                    // {
+                    //     case FriendshipEventResponse.BodyOneofCase.None: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Request: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Accept: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Reject: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Delete: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Cancel: break;
+                    //     default: throw new ArgumentOutOfRangeException();
+                    // }
+                }
+            }
+        }
+
+        public async UniTask RejectFriendship(string friendRequestId, CancellationToken cancellationToken = default)
+        {
+            string userId = GetUserIdFromFriendRequestId(friendRequestId);
+
+            var updateFriendshipPayload = new UpdateFriendshipPayload()
+            {
+                Event =
+                    new FriendshipEventPayload()
+                    {
+                        Reject = new RejectPayload()
+                        {
+                            User = new User() { Address = userId }
+                        }
+                    }
+            };
+
+            await this.UpdateFriendship(updateFriendshipPayload, cancellationToken);
+
+            OnFriendRequestRemoved?.Invoke(userId);
+        }
+
+        public async UniTask<FriendRequest> RequestFriendship(string friendUserId, string messageBody, CancellationToken cancellationToken = default)
+        {
+            long createdAt = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
```

Comment:
we may problems by calculating timestamps locally, since the user may have its own time desynchronized, similar to what happened to ~nearby messages

Reply:
That's not an issue, users will always have the ability to set their own custom times or even forge the HTTP request, unless we ask a server what time it is, there's no other way to do it

Reply:
It is going to be a different id when retrieve the friend request from the server, because the difference of the timestamp. So the friend request will be duplicated locally which may generate different issues. Perhaps the request to the server should retrieve the creation timestamp which will allow you to build the correct id